sdk/nodejs/apigateway/experimental/api.ts

```diff
@@ -69,17 +69,39 @@ export interface Response {
     body: string;
 }
 
-export type RouteEventHandler = lambda.EventHandler<Request, Response>;
 export type Method = "ANY" | "GET" | "PUT" | "POST" | "DELETE" | "PATCH";
-export type Route = {
-    path: string;
+
+/**
+ * A route that that APIGateway should accept and forward to some type of destination. All routes
+ * have an incoming path that they match against.  However, destinations are determined by the kind
+ * of the route.  See [EventHandlerRoute], [StaticRoute] and [ProxyRoute] for additional details.
+ */
+export type Route = EventHandlerRoute | StaticRoute | ProxyRoute;
+
+/**
+ * A mapping of [path]s to [Route]s.  The [path] is the incoming url path segment to match on.  The
+ * [route] is what behavior should be taken for that path.  See:
+ * https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-method-settings-method-request.html
+ * for more information on the types of paths allowed by apigateway.
+ */
+export type Routes = { [path: string]: Route };
+
+export type EventHandlerRoute = {
+    kind: "EventHandler";
```

Comment:
we coudl get rid of 'kind' if we're careful.  see first comment in PR.  Thoughts?